File: backend/src/main/java/com/apocalypse/thefall/exception/ErrorResponse.java
package com.apocalypse.thefall.exception;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Getter;
import org.springframework.http.HttpStatus;

import java.time.LocalDateTime;

@Getter
public class ErrorResponse {
    private final int status;
    private final String message;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd-MM-yyyy HH:mm:ss")
    private final LocalDateTime timestamp;

    public ErrorResponse(HttpStatus status, String message) {
        this.status = status.value();
        this.message = message;
        this.timestamp = LocalDateTime.now();
    }
}

File: backend/src/main/java/com/apocalypse/thefall/exception/ErrorResponse.kt
package com.apocalypse.thefall.exception

import com.fasterxml.jackson.annotation.JsonFormat
import org.springframework.http.HttpStatus
import java.time.LocalDateTime

data class ErrorResponse(
    val status: Int,
    val message: String,
    @field:JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd-MM-yyyy HH:mm:ss")
    val timestamp: LocalDateTime = LocalDateTime.now()
) {
    constructor(status: HttpStatus, message: String) : this(
        status = status.value(),
        message = message
    )
}
